certbot/lock.py

```diff
@@ -108,17 +164,101 @@ def _lock_success(self, fd):
         # the same device and inode, they're the same file.
         return stat1.st_dev == stat2.st_dev and stat1.st_ino == stat2.st_ino
 
-    def __repr__(self):
-        repr_str = '{0}({1}) <'.format(self.__class__.__name__, self._path)
-        if self._fd is None:
-            repr_str += 'released>'
-        else:
-            repr_str += 'acquired>'
-        return repr_str
-
     def release(self):
+        # type: () -> None
         """Remove, close, and release the lock file."""
+        # It is important the lock file is removed before it's released,
+        # otherwise:
+        #
+        # process A: open lock file
+        # process B: release lock file
+        # process A: lock file
+        # process A: check device and inode
+        # process B: delete file
+        # process C: open and lock a different file at the same path
         try:
-            compat.release_locked_file(self._fd, self._path)
+            os.remove(self._path)
         finally:
-            self._fd = None
+            try:
+                if self._fd:
```

Comment:
As commented below, yes, it is to make mypy happy. And again, release may be silently ignored when called a second time, when it was not the case before: do we raise an exception instead ?

Reply:
I think it's probably worth it. Keeps us from writing sloppy code.

Does it satisfy mypy to put that check at the top of the method like this?
```
if self._fd is None:
   raise Something
try:
    os.remove(self._path)
...
```
If `self._fd` is `None`, we probably should be deleting files.

Reply:
Here mypy complains, so I added an exception, that is what would throw os.close when None is given. Anyway the file is always removed, because os.remove is called before this line.